packages/wrangler/src/__tests__/dev.test.ts

```diff
@@ -458,6 +454,34 @@ describe.sequential("wrangler dev", () => {
 				Paths are not allowed in Custom Domains]
 			`);
 		});
+
+		it("should warn on mounted paths in dev", async () => {
+			writeWranglerConfig({
+				routes: [
+					"simple.co.uk/path/*",
+					"simple.co.uk/*",
+					{ pattern: "example.com/blog/*", zone_id: "asdfadsf" },
+					{ pattern: "example.com/*", zone_id: "asdfadsf" },
+					{ pattern: "example.com/abc/def/*", zone_id: "asdfadsf" },
+				],
+			});
+
+			fs.mkdirSync("assets");
+
+			await runWranglerUntilConfig("dev --assets assets");
+
+			expect(std.warn).toMatchInlineSnapshot(`
+				"[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mWarning: The following routes will attempt to serve Assets on a mounted path:[0m
+
+				    • simple.co.uk/path/* (Will match assets: assets/path/*)
+				    • example.com/blog/* (Will match assets: assets/blog/*)
+				    • example.com/abc/def/* (Will match assets: assets/abc/def/*)
+
+				  Requests not matching an asset will be forwarded to the Worker.
```

Comment:
```suggestion
				  Requests not matching an asset will be forwarded to the Worker script.
```

Reply:
Do we want to say script? We moved away from the script terminology a while ago... though assets are part of the worker so also understandable

Reply:
We're not perfectly consistent, but yes, I'm trying to not make it seem like assets are "outside" of the Worker. We've used "script" in the docs for this a few times: https://developers.cloudflare.com/workers/static-assets/binding/.

We originally turned away from "script" because "service" replaced it. That's not really the same issue here, but yeah, totally get your concern. If you've got a better way of phrasing it, I'm all ears!